feg/gateway/services/eap/client/client_api.go

```diff
@@ -57,17 +57,26 @@ func HandleIdentityResponse(providerType uint8, msg *protos.Eap) (*protos.Eap, e
 		}
 	}
 	var p providers.Method
-	if providerType == 0 && len(td) > 0 { // no EAP type specified, find a provider willing to handle the Identity
-		for _, typ := range supportedEapTypes {
-			pt := registry.GetProvider(typ)
-			if pt != nil && pt.WillHandleIdentity(td) {
-				p = pt
-				break
+	if providerType == 0 { // no EAP type specified, find a provider willing to handle the Identity
+		if len(td) > 0 {
+			for _, typ := range supportedEapTypes {
+				pt := registry.GetProvider(typ)
+				if pt != nil && pt.WillHandleIdentity(td) {
+					p = pt
+					break
+				}
```

Comment:
so does this mean we can have different providers running at the same time?

Reply:
yes, all current providers can identify a matching user identity by the identity format